app/gui2/src/components/visualizations/ScatterplotVisualization.vue

```diff
@@ -132,35 +144,71 @@ const SHAPE_TO_SYMBOL: Record<string, d3.SymbolType> = {
   triangle: d3.symbolTriangle,
 }
 
-const SCALE_TO_D3_SCALE: Record<ScaleType, () => d3.ScaleContinuousNumeric<number, number>> = {
+const SCALE_TO_D3_SCALE: any = {
   [ScaleType.Linear]: () => d3.scaleLinear(),
   [ScaleType.Logarithmic]: () => d3.scaleLog(),
+  [ScaleType.Time]: () => d3.scaleTime(),
+}
+
+const createDateTime = (x: DateObj) => {
+  const dateTime = new Date()
+  x.day ? dateTime.setDate(x.day) : dateTime
```

Comment:
This is a wierd use for a ternary. Those could be simple if expressions. Also they don't check for falsy non-nullable values (e.g. `0`).
```suggestion
  if(x.day != null) dateTime.setDate(x.day)
```